inc/css/class-block-frontend.php

```diff
@@ -0,0 +1,433 @@
+			foreach( $fonts_list as $font ) {
+				$item = str_replace( ' ', '+', $font['fontfamily'] );
+				if ( sizeof( $font['fontvariant'] ) > 0 ) {
+					$item .= ':' . implode( ',', $font['fontvariant'] );
+				}
+				array_push( $fonts, $item );
+			}
+
+			wp_enqueue_style( 'themeisle-gutenberg-google-fonts', '//fonts.googleapis.com/css?family=' . implode( '|', $fonts ) );
+		}
+	}
+
+	/**
+	 * Method to define hooks needed.
+	 * 
+	 * @since   1.2.5
+	 * @access  public
+	 */
+	public function enqueue_google_fonts_backward() {
+		if ( $this->has_fonts ) {
+			return;
+		}
+
+		$fonts = array();
+
+		if ( sizeof( self::$google_fonts ) > 0 ) {
+			foreach( self::$google_fonts as $font ) {
+				$item = str_replace( ' ', '+', $font['fontfamily'] );
+				if ( sizeof( $font['fontvariant'] ) > 0 ) {
+					$item .= ':' . implode( ',', $font['fontvariant'] );
+				}
+				array_push( $fonts, $item );
+			}
+
+			wp_enqueue_style( 'themeisle-gutenberg-google-fonts', '//fonts.googleapis.com/css?family=' . implode( '|', $fonts ) );
+		}
+	}
+
+	/**
+	 * Get Google Fonts for Reusable Blocks
+	 * 
+	 * @since   1.2.5
+	 * @access  public
+	 */
+	public function enqueue_reusable_fonts( $blocks ) {
+		foreach ( $blocks as $block ) {
+			if ( $block['blockName'] === 'core/block' && ! empty( $block['attrs']['ref'] ) ) {
+				$this->enqueue_google_fonts( $block['attrs']['ref'] );
+			}
+
+			if ( isset( $block['innerBlocks'] ) && ! empty( $block['innerBlocks'] ) && is_array( $block['innerBlocks'] ) ) {
+				$this->enqueue_reusable_fonts( $block['innerBlocks'] );
+			}
+		}
+	}
+
+	/**
+	 * Render server-side CSS
+	 * 
+	 * @since   1.2.5
+	 * @access  public
+	 */
+	public function render_post_css() {
+		if ( is_singular() ) {
+			return $this->enqueue_styles();
+		}
+
+		add_filter( 'the_content', function( $content ) {
+			if ( $this->has_excerpt ) {
+				return $content;
+			}
+
+			$post_id = get_the_ID();
+
+			$this->enqueue_styles( $post_id, true );
+
+			return $content;
+		} );
+	}
+
+	/**
+	 * Enqueue CSS file
+	 * 
+	 * @since   1.2.5
+	 * @access  public
+	 */
+	public function enqueue_styles( $post_id = '', $footer = false ) {
+		$post_id = $post_id ? $post_id : get_the_ID();
+		$location = 'wp_head';
+
+		if ( function_exists( 'has_blocks' ) && has_blocks( $post_id ) ) {
+			$file_name = get_post_meta( $post_id, '_themeisle_gutenberg_block_stylesheet', true );
+
+			if ( $footer ) {
+				$location = 'wp_footer';
+			}
+
+			if ( empty( $file_name ) || is_preview() ) {
+				return add_action( $location, function() use( $post_id ) {
+					return $this->get_post_css( $post_id );
+				} );
+			}
+
+			$wp_upload_dir = wp_upload_dir( null, false );
+			$basedir = $wp_upload_dir['basedir'] . '/themeisle-gutenberg/';
+			$baseurl = $wp_upload_dir['baseurl'] . '/themeisle-gutenberg/';
+			$file_path = $basedir . $file_name . '.css';
+			$file_url = $baseurl . $file_name . '.css';
+
+			if ( ! file_exists( $file_path ) ) {
```

Comment:
if the file does not exist, you should bail also, i see here you also include it https://github.com/Codeinwp/gutenberg-blocks/pull/119/files#diff-82cbf82ffb99989c410a08cf5907d422R256

Reply:
I don't understand what you mean here. Can you explain?

Reply:
@selul Can you explain this?

Reply:
not sure if you checked by reference, but I'm adding here a screenshot http://snpy.in/JN2oP5

In the outlined line, you enqueue the file, even if it exists or not.

Reply:
If the file doesn't exist, we enqueue the CSS to footer directly instead of the file.